projects/Mallard/src/hooks/use-image-paths.ts

```diff
@@ -98,17 +102,41 @@ export const useImagePath = (image?: Image, use: ImageUse = 'full-size') => {
     return path
 }
 
+type QueryValue = { scaledImage: { uri: string | null } }
+type QueryVars = { path: string; width: number }
+const SCALED_IMAGE_QUERY = gql`
+    query($path: String, $width: Int) {
+        scaledImage(path: $path, width: $width) @client
+    }
+`
+
+export const createScaledImageResolver = () => {
+    const promises = new Map()
```

Comment:
is `promises` the best name here? maybe 'compressedImagePaths' or something?

Reply:
yeah will fix